server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -905,6 +909,19 @@ public void close(boolean saveState) throws IOException {
   void initiateClose(boolean saveState) {
     log.trace("initiateClose(saveState={}) {}", saveState, getExtent());
 
+    synchronized (this) {
+      if (closeState == CloseState.OPEN) {
+        closeRequestTime = System.nanoTime();
+      } else if (closeRequestTime != 0) {
+        long runningTime = Duration.ofNanos(System.nanoTime() - closeRequestTime).toMinutes();
+        if (runningTime >= 15) {
+          DEDUPE_LOGGER.info("Tablet {} close requested again, but has been closing for {} minutes",
+              this.extent, runningTime);
+        }
+      }
+      closeState = CloseState.REQUESTED;
```

Comment:
Only want to transition to REQUESTED when the current state is OPEN.  A thread could call this method when the close state has advanced past REQUESTED.


```suggestion
      if (closeState == CloseState.OPEN) {
        closeRequestTime = System.nanoTime();
        closeState = CloseState.REQUESTED;
      } else if (closeRequestTime != 0) {
        long runningTime = Duration.ofNanos(System.nanoTime() - closeRequestTime).toMinutes();
        if (runningTime >= 15) {
          DEDUPE_LOGGER.info("Tablet {} close requested again, but has been closing for {} minutes",
              this.extent, runningTime);
        }
      }
```

Reply:
Change made in 3fa4f87